packages/flutter/lib/src/animation/animation_controller.dart

```diff
@@ -195,6 +195,82 @@ enum AnimationBehavior {
 /// ```
 /// {@end-tool}
 ///
+/// {@tool dartpad --template=freeform}
+///
+/// ```dart imports
+/// import 'package:flutter/material.dart';
+/// ```
+///
+/// ```dart
+/// void main() {
+///   runApp(AnimationControllerDemo());
+/// }
+///
+/// class AnimationControllerDemo extends StatefulWidget {
+///   @override
+///   _AnimationControllerDemoState createState() =>
+///       _AnimationControllerDemoState();
+/// }
+///
+/// class _AnimationControllerDemoState extends State<AnimationControllerDemo>
+///     with SingleTickerProviderStateMixin {
+///   static const Duration _duration = Duration(seconds: 1);
+///   dynamic controller;
+///
+///   @override
+///   void initState() {
+///     super.initState();
+///     controller = AnimationController(vsync: this, duration: _duration)
+///       ..addListener(() {
+///         setState(() {});
+///       });
```

Comment:
An empty setState call usually isn't idiomatic. You'd probably want to use an AnimatedBuilder to only rebuild the part of the tree that depends on the controller value.